### PR TITLE
Add selectionColor prop for Text on Android

### PR DESCRIPTION
**Pull request by satya164**
**Motivation**

Customizing the selection color allows to use brand colors in the app. The PR implements a `selectionColor` prop for `Text` component similar to `TextInput`.

**Test plan (required)**

Run UIExplorer example with the changes and verify everything works fine.

image

cc @brentvatne

### CODE REVIEW

File: Libraries/Text/Text.js [hunk 1]
```diff
@@ -32,6 +36,7 @@ const viewConfig = {
     ellipsizeMode: true,
     allowFontScaling: true,
     selectable: true,
+    selectionColor: ColorPropType,
```
Comment (janicduplessis) [inline Libraries/Text/Text.js]: I think this should be `selectionColor: true`

File: Libraries/Text/Text.js [hunk 2]
```diff
@@ -317,6 +326,12 @@ const Text = React.createClass({
         isHighlighted: this.state.isHighlighted,
       };
     }
+    if (typeof newProps.selectionColor !== 'undefined') {
```
Comment (janicduplessis) [inline Libraries/Text/Text.js]: Looking at other examples (TextInput) it seems like we don't need to manually call processColor for props of type Color.

File: Libraries/Text/Text.js (hunk 1 above)
Comment (satya164) [inline Libraries/Text/Text.js]: @janicduplessis you're right. I got confused by another place where there was proptype, most probably by mistake.

File: Libraries/Text/Text.js (hunk 2 above)
Comment (satya164) [inline Libraries/Text/Text.js]: @janicduplessis `Text` doesn't use the `requireNativeComponent` which calls `processColor`, so this seems to be needed.

Comment (janicduplessis) [inline Libraries/Text/Text.js]: Ahh I see, Could you just change the if check to be newProps.selectionColor != null instead to match null and undefined.

**Comment by satya164** [comment]
@janicduplessis updated

File: Libraries/Text/Text.js [hunk 3]
```diff
@@ -11,6 +11,8 @@
  */
 'use strict';
 
+const ColorPropType = require('ColorPropType');
+const processColor = require('processColor');
```
Comment (janicduplessis) [inline Libraries/Text/Text.js]: Nit: Move this with other function requires (those that start with a lowercase letter)

File: Libraries/Text/Text.js [hunk 4]
```diff
@@ -153,24 +158,28 @@ const Text = React.createClass({
     /**
      * Lets the user select text, to use the native copy and paste functionality.
      */
-    selectable: React.PropTypes.bool,
+    selectable: PropTypes.bool,
+    /**
+    * The highlight color of the text.
```
Comment (janicduplessis) [inline Libraries/Text/Text.js]: Nit: Align comment `*`s.

**Comment by satya164** [comment]
@janicduplessis updated

**Comment by janicduplessis** [comment]
👍 Will ship later today once CI passes.

**Comment by janicduplessis** [comment]
@facebook-github-bot shipit

File: ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextViewManager.java [hunk 5]
```diff
@@ -96,6 +96,15 @@ public void setSelectable(ReactTextView view, boolean isSelectable) {
     view.setTextIsSelectable(isSelectable);
   }
 
+  @ReactProp(name = "selectionColor", customType = "Color")
+  public void setSelectionColor(ReactTextView view, @Nullable Integer color) {
+    if (color == null) {
+      view.setHighlightColor(DefaultStyleValuesUtil.getDefaultTextColorHighlight(view.getContext()));
```
Comment (mkonicek) [inline ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextViewManager.java]: nit: remove empty line

**Comment by janicduplessis** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**